src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -855,24 +912,25 @@ def list_daemons(self, service_name=None, daemon_type=None, daemon_id=None, host
         """
         raise NotImplementedError()
 
-    def apply(self, specs: List[ServiceSpec]) -> Completion:
+    def apply(self, specs: List[GenericSpec]) -> Completion:
```

Comment:
you can forward-declare things by using strings:

```suggestion
    def apply(self, specs: List["GenericSpec"]) -> Completion:
```

this way you don't need to move HostSpec